Userland/Libraries/LibURL/Parser.cpp

```diff
@@ -304,9 +304,8 @@ static void serialize_ipv6_address(IPv6Address const& address, StringBuilder& ou
         if (ignore0 && address[piece_index] == 0)
             continue;
 
-        // 2. Otherwise, if ignore0 is true, set ignore0 to false.
-        if (ignore0)
-            ignore0 = false;
+        // 2. Set ignore0 to false.
+        ignore0 = false;
```

Comment:
The comment was a direct quote from the spec. Please don't change it.

Reply:
> LibURL: Fixed a case of ignore0 being checked before assignment

That isn't the case here though? It's initialized just a couple lines above here.
```c++
// 4. Let ignore0 be false.
auto ignore0 = false;
```

Reply:
> > LibURL: Fixed a case of ignore0 being checked before assignment
> 
> That isn't the case here though? It's initialized just a couple lines above here.
> 
> ```c++
> // 4. Let ignore0 be false.
> auto ignore0 = false;
> ```

Please mind that I am just removing the redundant check for `ignore0`.